be/src/vec/functions/function_string.h

```diff
@@ -3155,9 +3155,14 @@ class FunctionReplace : public IFunction {
 
         for (int i = 0; i < input_rows_count; ++i) {
             StringRef origin_str =
-                    assert_cast<const ColumnString*>(col_origin.get())->get_data_at(i);
-            StringRef old_str = assert_cast<const ColumnString*>(col_old.get())->get_data_at(i);
-            StringRef new_str = assert_cast<const ColumnString*>(col_new.get())->get_data_at(i);
+                    assert_cast<const ColumnString*, TypeCheckOnRelease::DISABLE>(col_origin.get())
+                            ->get_data_at(i);
+            StringRef old_str =
+                    assert_cast<const ColumnString*, TypeCheckOnRelease::DISABLE>(col_old.get())
+                            ->get_data_at(i);
+            StringRef new_str =
+                    assert_cast<const ColumnString*, TypeCheckOnRelease::DISABLE>(col_new.get())
+                            ->get_data_at(i);
```

Comment:
`assert_cast` can be moved out before `for` loop and enable type check.

Reply:
fixed